zkevm-circuits/src/util.rs

```diff
@@ -34,3 +36,62 @@ pub fn power_of_randomness_from_instance<F: Field, const N: usize>(
 
     power_of_randomness.unwrap()
 }
+
+/// All challenges used in `SuperCircuit`.
+#[derive(Clone, Copy, Debug)]
+pub struct Challenges<T = Challenge> {
+    evm_word: T,
+    keccak_input: T,
+}
+
+impl Challenges {
+    /// Construct `Challenges` by allocating challenges in specific phases.
+    pub fn construct<F: Field>(meta: &mut ConstraintSystem<F>) -> Self {
+        Self {
+            evm_word: meta.challenge_usable_after(FirstPhase),
+            keccak_input: meta.challenge_usable_after(FirstPhase),
+        }
+    }
+
+    /// Get `Expression`s from `Layouter` and returns expressions of challenges.
+    pub fn exprs<F: Field>(&self, meta: &mut ConstraintSystem<F>) -> Challenges<Expression<F>> {
+        let mut challenges = array::from_fn(|_| None);
+        meta.create_gate("power of randomness from instance", |meta| {
```

Comment:
1. Does "power of randomness from instance" describe what we are doing here?
2. Maybe not in this PR but in a future one, is there a way to abstract away this create_gate trick to get expressions from ConstraintSystem?

Reply:
1. No, I copied but forgot to update the description, thanks for catching this!
2. Yes, let me add a wrapper do that.

Reply:
Fixed in https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/770/commits/96fa4e253365a556fec718ed3550c568f5b4dbd5, and add a helper `query_expression` to avoid create dummy gate manually.